### PR TITLE
S3

### PR DESCRIPTION
**Pull request by habdelra**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by bgrieder** [comment]
👍

**Comment by vvakame** [comment]
We can't get response from author. please ping me after 3 days. I'll merge this PR.

**Comment by midknight41** [comment]
@habdeira I don't think this one is right. copyObject and getObject do not have the same request params.

File: aws-sdk/aws-sdk.d.ts [hunk 1]
```diff
@@ -382,6 +382,7 @@ declare module "aws-sdk" {
     endpoint: Endpoint;
 
     getObject(params: s3.GetObjectRequest, callback?: (err: Error, data: any) => void): any;
+    copyObject(params: s3.CopyObjectRequest, callback?: (err: Error, data: any) => void): any;
```
Comment (habdelra) [inline aws-sdk/aws-sdk.d.ts]: @midknight41 I created a separate request object for copy

**Comment by midknight41** [comment]
Apologies, I missed your commit. 

I think that metadata is better described like:

```Metadata?: { [key: string]: string; };```

Outside of that, it looks fine.

**Comment by habdelra** [comment]
@midknight41 thanks, I added the metadata update you suggested

**Comment by midknight41** [comment]
:+1:

**Pull request merged**